src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -200,32 +254,65 @@ private static WellKnownMember GetTupleTypeMember(int arity, int position)
                                                             WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item4,
                                                             WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item5,
                                                             WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item6,
-                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item7 }
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item7 },
+
+                                                        new[]{
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item1,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item2,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item3,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item4,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item5,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item6,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item7,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Rest }
         };
 
         /// <summary>
         /// Returns "Item1" for position=0
+        /// Returns "Item12" for position=11
         /// </summary>
-        internal static string UnderlyingMemberName(int position)
+        internal static string TupleMemberName(int position)
```

Comment:
Consider if it would make sense to make TupleMemberName and IsMemberNameReserved consistent in terms of position numbers. IsMemberNameReserved uses one-based positions, TupleMemberName uses zero-based positions.